### DISCUSSION TITLE
Is there a test suite for output?

### DISCUSSION THREAD

**Discussion by lockenj** [json-schema-org/community#387] [Q&A] [answered]
I have found the test suite (https://github.com/json-schema-org/JSON-Schema-Test-Suite) insanely valuable for working through validations... 

I am now attempting to implement output (annotations, errors) inline with what I believe is the agreed go forward approach, detailed here https://json-schema.org/blog/posts/fixing-json-schema-output. 

I have been leveraging https://json-everything.net/json-schema/ however I'd love a complete suite that I can program against.

I'd be extremely grateful for any guidance!

Thanks @gregsdennis @handrewsfor putting out so much helpful content... It has helped me tremendously.

**Comment by Julian** [comment, marked as answer]
Hey! What we have so far lives [here](https://github.com/json-schema-org/JSON-Schema-Test-Suite/tree/main/output-tests), though there's also some discussion ongoing [here](https://github.com/json-schema-org/JSON-Schema-Test-Suite/issues/657) which could definitely use your input or contributions!

**Reply by lockenj** [reply]
@Julian that was super helpful I appreciate that. Using the above I was able to quickly implement output test for the draft-next output.

**Comment by lockenj** [comment]
In case someone else comes along and is also looking to understand the current draft-next output...

In addition to the links @Julian provided above I found it extremely helpful to clone the [json-schema-spec repo](https://github.com/json-schema-org/json-schema-spec/tree/main) and run the makefile so you can read the draft-next spec. Finally this ([article](https://json-schema.org/blog/posts/fixing-json-schema-output)) was great for an overview of what is changing.

**Reply by gregsdennis** [reply]
There may be additional changes coming as well.  For now, what's in the document is what we have, so that's what the tests are checking.  Potential changes won't change the structure, but would rather add some additional properties to the root of the output.  We don't have anything finalized yet, though.  Check the `output` label in the spec repo for discussions.
